internal/pkg/api/handleEnroll.go

```diff
@@ -252,6 +261,81 @@ func (et *EnrollerT) _enroll(
 				Msg("Error when trying to delete old agent with enrollment id")
 			return nil, err
 		}
+		// deleted, so clear the ID so code below knows it needs to be created
+		agent.Id = ""
+	}
+
+	var agentID string
+	if req.Id != nil && *req.Id != "" {
+		agentID = *req.Id
+
+		// check if the agent with this ID already exists
+		var err error
+		agent, err = et._checkAgent(ctx, zlog, agentID)
+		if err != nil {
+			return nil, err
+		}
+
+		if agent.Id != "" {
```

Comment:
can it return `agent.id == ""`? if not this if statement is not needed.
if so. we dont have this case handled as this should not be the same as empty ID when `req.ID` is not used.
we whould probably not continue with empty id, probably we should fail. generating a new one breaks the purpose of providing it via req.id

Reply:
It absolutely will return an `agent.id == ""`. This happens when we check to see if an existing agent already exists with that ID. The ErrNotFound will not be returned from `_checkAgent`, it will return `nil` error and this will be `agent.id == "".